TESTING.md

```diff
@@ -0,0 +1,67 @@
+# Testing the 'oneview' module
+The 'oneview' module provides a unit testing strategy and linting tools to help ensure its code style.
+This document will cover on how to execute and implement these tests.
+
+## Executing tests
+The tests can be executed by `rake` tasks and by their original tools, like `rspec`, `rubocop`, `puppet-lint` and `puppet parser`.
+
+### Rake
+All the test strategies and checks can be executed by the rake command.
+Please use `rake -T` to see a full list of commands.
+
+### Unit & Integration tests
+All unit tests are inside the spec folder. You can execute them manually by using `rspec` and specifying the test files, like `rspec spec/path/to/my/tests`.
+If `bundler` was used to install the tools and test suite, which is a good practice, it is also recommended to run the unit tests prefixed by `bundle exec`, so that the correct gems and versions are used.
+
+### Functional tests
+Every resource must have an example for each action that shows how to use it. These examples can be used to perform functional and OneView integration tests but they need to be executed manually using the `puppet apply <manifest.pp>` command.
+
+#### Coverage
+The unit tests issue a coverage report generated by `SimpleCov` and validated with `Coveralls`.
+
+### Style checking
+Style checking is performed by:
+- `rubocop` for all the custom resource providers and types
+- `puppet-lint` and `puppet parser` for manifests.
+
+## Implementing tests
+All code must have tests associated with it, and this section covers what tests need to be implemented.
+
+### Unit tests
+The unit tests are required for each custom provider and type methods.
+
+### Integration tests
```

Comment:
Aren't they the examples? If so, I recommend calling them examples and mentioning they have the work as Integration tests too.

Reply:
Nope, there are integration tests and the `examples` are covered in `Functional` tests.